src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -757,6 +809,89 @@ public void reset() {
         return mDeviceInfo;
     }
 
+    /**
+     * Use this method to opt-out a user from tracking. Events and people updates that haven't been
+     * flushed yet will be deleted. Use {@link #flush()} before calling this method if you want
+     * to send all the queues to Mixpanel before.
+     *
+     * This method will also remove any user-related information from the device.
+     */
+    public void optOutTracking() {
+        getAnalyticsMessages().emptyTrackingQueues(new AnalyticsMessages.MixpanelDescription(mToken));
+        if (getPeople().isIdentified()) {
+            getPeople().deleteUser();
+            getPeople().clearCharges();
+        }
+        mPersistentIdentity.clearPreferences();
+        synchronized (mEventTimings) {
+            mEventTimings.clear();
+            mPersistentIdentity.clearTimeEvents();
+        }
+        mPersistentIdentity.clearReferrerProperties();
+        mPersistentIdentity.setOptOutTracking(true);
```

Comment:
do we want to call `unregisterSuperProperty` as well?

Reply:
`mPersistentIdentity.clearPreferences` removes all the super properties